public/js/actions/breakpoints.js

```diff
@@ -22,15 +23,51 @@ function enableBreakpoint(location: Location) {
   return addBreakpoint(location);
 }
 
+function isOriginalLocation(state, location: Location) {
+  const source = getSource(state, location.sourceId);
+  return isOriginal(source.toJS());
+}
+
+function getGeneratedLocation(state, location: Location) {
+  const source = getSource(state, location.sourceId);
+  if (isOriginal(source.toJS())) {
+    return getGeneratedSourceLocation(source.toJS(), location);
+  }
+
+  return location;
+}
+
 function _breakpointExists(state, location: Location) {
+  location = getGeneratedLocation(state, location);
   const currentBp = getBreakpoint(state, location);
   return currentBp && !currentBp.get("disabled");
 }
 
 function _getOrCreateBreakpoint(state, location, condition) {
+  location = getGeneratedLocation(state, location);
   return getBreakpoint(state, location) || fromJS({ location, condition });
 }
 
+function getBreakpointText({
+  breakpoint, getState, getTextForLine, actualLocation, location }) {
+  // If this breakpoint is being re-enabled, it already has a
+  // text snippet.
+  let text = breakpoint.get("text");
+  if (text) {
+    return text;
+  }
+
+  if (!getTextForLine) {
+    return "";
+  }
+
+  if (isOriginalLocation(getState(), location)) {
```

Comment:
Since many point of code will have to deal with sourcemapping, I think it's easier to do it like the current debugger server: have a function like `getOriginalLocation` that takes a location and always returns the original location. If the source is not sourcemapped, it'll just return the same location. If it is, it'll sourcemap it.

But it may be you are still working through this :) I don't quite understand this code because `location` and `actualLocation` are not different here.